source/common/router/config_impl.cc

```diff
@@ -1118,8 +1118,10 @@ RouteConstSharedPtr RouteEntryImplBase::pickWeightedCluster(const Http::HeaderMa
       if (absl::SimpleAtoi(header_value[0]->value().getStringView(), &random_value)) {
         random_value_from_header = random_value;
       }
-    } else {
-      // Random value should be found here. But if it is not found due to some errors, log the
+    }
+
+    if (!random_value_from_header.has_value()) {
```

Comment:
I think this will generate a log line for use-cases where cluster picking isn't configured to use a random-value-header.
In this case we should probably reduce the log-level to trace (less spam), or make sure that it is only executed when random-value-header is configured.

Reply:
This is inside of the if check `if (!random_value_header_name_.empty()) `. So this is only for the case that cluster picking is configured to use a random-value-header.

Reply:
Ah, you are correct, my bad.